lib/logproto/logproto-server.c

```diff
@@ -122,12 +122,20 @@ log_proto_server_init(LogProtoServer *self, LogTransport *transport, const LogPr
   self->transport = transport;
 }
 
-void
+gboolean
 log_proto_server_options_set_encoding(LogProtoServerOptions *self, const gchar *encoding)
 {
-  if (self->encoding)
-    g_free(self->encoding);
+  GIConv convert;
+
+  g_free(self->encoding);
   self->encoding = g_strdup(encoding);
+
+  /* validate encoding */
+  convert = g_iconv_open("utf8", encoding);
```

Comment:
@bkil-syslogng : my main problem is the way how you communicated all this staff.
I won't repeat myself (key point: share all the information you have such a case like this and don't let others working on an issue that is already resolved).

Some technical details:
 * glib can use the libiconv or the glibc iconv when it is available

The glibc iconv accepts UTF-8, utf-8, UTF8 and even utf8 but the libiconv implementation
does not accept encoding strings without '-'.

We agreed on that we will open a github issue and @bkil-syslogng will send a fix.
The solution is that we won't use 'utf8' but 'utf-8'...